tasks/main.yml

```diff
@@ -3,5 +3,13 @@
 - name: Set ansible_facts required by role
   include_tasks: set_vars.yml
 
+- name: Handle insights unregistration
+  include_tasks: insights-client-unregistration.yml
+  when: rhc_insights.state | d('present') == "absent"
+
 - name: Handle system subscription
   include_tasks: subscription-manager.yml
+
+- name: Handle insights registration
+  include_tasks: insights-client.yml
+  when: rhc_insights.state | d('present') == "absent"
```

Comment:
most likely, the comparison here ought to be with "present" rather than "absent" ;)

also, as I mentioned for the above `Handle insights unregistration task:
- perform this task only on RHEL
- minor nitpick: double quotes instead of single ones